boa/environment.py

```diff
@@ -340,3 +341,43 @@ def time_travel(
 
         self.evm.patch.timestamp += seconds
         self.evm.patch.block_number += blocks
+
+    @cached_property
+    def block(self) -> "BlockEnv":
+        return BlockEnv(self.evm.patch)
+
+
+class MagicAttribute:
+    def __init__(self, patch_attr: str):
+        self.patch_attr = patch_attr
+
+    def __set_name__(self, owner: Type["BlockEnv"], name: str) -> None:
+        ...  # python descriptor protocol
+
+    def __get__(self, block: "BlockEnv", owner):
+        if block is None:
+            return self  # static call
+        attr = self.patch_attr
+        patch = block.patch
+        value = getattr(patch, attr)
+        base: Type = type(value)
+
+        class MakeCallable(base):
```

Comment:
What is this magic :o

Reply:
so dirty :scream: 

it allows you to call the same property as `get`, `set` or `call`: https://github.com/vyperlang/titanoboa/pull/223/files#diff-ffbd38618512c3ebf87ee5a7a54efc7c32c86c811a6792dd8f6c87aadd76960dR28-R33

Reply:
another approach i thought about is subclassing `int`, and adding a `__call__()` implementation. so you do `with number(5): ...`

also dirty!

Reply:
tbh I think this is not great. It gets IDEs and mypy (and therefore other devs) very confused